src/libNetwork/P2PComm.cpp

```diff
@@ -469,16 +469,18 @@ void P2PComm::ClearBroadcastHashAsync(const bytes& message_hash) {
       // Queue the message
       m_dispatcher(raw_message);
     }
-  } else if (p2p.m_rumorManager.RumorReceived((unsigned int)gossipMsgTyp,
-                                              gossipMsgRound, rumor_message,
-                                              from)) {
-    std::pair<bytes, Peer>* raw_message =
-        new pair<bytes, Peer>(rumor_message, from);
+  } else {
+    auto resp = p2p.m_rumorManager.RumorReceived(
+        (unsigned int)gossipMsgTyp, gossipMsgRound, rumor_message, from);
+    if (resp.first) {
+      std::pair<bytes, Peer>* raw_message =
```

Comment:
can we try to use smart pointer here?

Reply:
Tried to reuse the existing approach that we infact convert the smart_pointer to raw pointer before passing on to dispatcher 
https://github.com/Zilliqa/Zilliqa/blob/d6e7a2a24c5c56283b671d73a5db08782569febf/src/libNetwork/P2PComm.cpp#L606
Let me know your view.